src/xcode/ENA/ENA/Source/Services/HealthCertificate/Models/Extensions/HealthCertificateArray+MostRelevant.swift

```diff
@@ -42,18 +42,34 @@ extension Array where Element == HealthCertificate {
 	}
 
 	var lastCompleteVaccinationCertificate: HealthCertificate? {
-		last {
+		// 1- we filter all the certificates to get an array with only complete Vaccination Certificates
+		let completeVaccinationCertificates = filter {
 			guard let vaccinationEntry = $0.vaccinationEntry else {
 				return false
 			}
-
+			
 			return vaccinationEntry.doseNumber >= vaccinationEntry.totalSeriesOfDoses && (
 				$0.ageInDays ?? 0 > 14 ||
 				vaccinationEntry.isBoosterVaccination ||
-				vaccinationEntry.isRecoveredVaccination)
+				vaccinationEntry.isRecoveredVaccination ||
+				vaccinationEntry.doseNumber > vaccinationEntry.totalSeriesOfDoses
+			)
+		}
+		guard let latestCompleteVaccinationCertificateAgeInDays = completeVaccinationCertificates.last?.ageInDays else {
+			return nil
+		}
+		
+		// 2- check if there is multiple complete vaccination certificates that have the same ageInDays
+		let completeVaccinationCertificatesWithTheSameAgeInDays = completeVaccinationCertificates.filter {
+			$0.ageInDays == latestCompleteVaccinationCertificateAgeInDays
 		}
+		
+		// 3- return the certificate with the latest issuedAt date
+		return completeVaccinationCertificatesWithTheSameAgeInDays.sorted(by: {
+			$0.cborWebTokenHeader.issuedAt < $1.cborWebTokenHeader.issuedAt
+		}).max()
```

Comment:
Instead of checking for same-day vaccinations I would propose to enhance the `Comparable` implementation of the `HealthCertificate` class. You can introduce a secondary order by using tuples like this: `(lhsDate, lhs.cborWebTokenHeader.issuedAt) < (rhsDate, lhs.cborWebTokenHeader.issuedAt)`. We are already sorting certificates when the are added to a person, so with this enhanced implementation the should already be in the right order when the most relevant certificate is determined in this extension.

With this approach it would also be preferable to have unit tests in the tests for the `HealthCertificate` class. I think there should already be some that test the sorting that you can maybe adapt to check for same-day or even same-time certificates.